lib/storage/data/file/DataFileStore.js

```diff
@@ -193,7 +193,17 @@ class DataFileStore {
                     if (this.noCache) {
                         releasePageCacheSync(filePath, fd, log);
                     }
-                    fs.close(fd);
+                    const reqUid = log.getSerializedUids();
+                    fs.close(fd, err => {
+                        if (err) {
+                            this.logger.error('error closing fd after write', {
```

Comment:
Don't you want to do `log.error`?

Reply:
Since we are not waiting for callback there will be a chance when end logger is called on `log` and if this call happens after that then an exception will be thrown. So, I am using the module logger for safety and logging the context.